clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp

```diff
@@ -0,0 +1,49 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-host -triple -x86_64-unknown-linux-gnu -disable-llvm-passes %s -emit-llvm -o -  | FileCheck %s
+
+class Functor16 {
+public:
+  [[intel::reqd_sub_group_size(16)]] void operator()() const {}
+};
+
+[[intel::reqd_sub_group_size(8)]] void foo() {}
+
+class Functor {
+public:
+  void operator()() const {
+    foo();
+  }
+};
+
+template <int SIZE>
+class Functor5 {
+public:
+  [[intel::reqd_sub_group_size(SIZE)]] void operator()() const {}
+};
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel(const Func &kernelFunc) {
+  kernelFunc();
+}
```

Comment:
Could you please use queue::submit and handler::single task from the mock header instead? (Like here https://github.com/intel/llvm/blob/sycl/clang/test/CodeGenSYCL/stall_enable.cpp). We need to keep single style in FE tests.

Reply:
@Fznamznon This single style does not work on host (-fsycl-is-host). I tried this and it could not find the path and failed with lots of errors like this:
Command Output (stderr):
--
In file included from /export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp:3:
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:190:1: error: unknown type name '__ocl_image1d_ro_t'
IMAGETY_READ_3_DIM_IMAGE
^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:173:3: note: expanded from macro 'IMAGETY_READ_3_DIM_IMAGE'
  IMAGETY_DEFINE(1, read, ro, image, ) \
  ^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:169:18: note: expanded from macro 'IMAGETY_DEFINE'
    using type = __ocl_image##dim##d_##ifarray_##amsuffix##_t;      \
                 ^
<scratch space>:6:1: note: expanded from here
__ocl_image1d_ro_t
^
In file included from /export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp:3:
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:190:1: error: unknown type name '__ocl_image2d_ro_t'
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:174:3: note: expanded from macro 'IMAGETY_READ_3_DIM_IMAGE'
  IMAGETY_DEFINE(2, read, ro, image, ) \
  ^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:169:18: note: expanded from macro 'IMAGETY_DEFINE'
    using type = __ocl_image##dim##d_##ifarray_##amsuffix##_t;      \
                 ^
<scratch space>:10:1: note: expanded from here
__ocl_image2d_ro_t
^
In file included from /export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp:3:
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:190:1: error: unknown type name '__ocl_image3d_ro_t'
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:175:3: note: expanded from macro 'IMAGETY_READ_3_DIM_IMAGE'
  IMAGETY_DEFINE(3, read, ro, image, )
  ^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:169:18: note: expanded from macro 'IMAGETY_DEFINE'
    using type = __ocl_image##dim##d_##ifarray_##amsuffix##_t;      \
                 ^
<scratch space>:14:1: note: expanded from here
__ocl_image3d_ro_t
^
In file included from /export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp:3:
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:191:1: error: unknown type name '__ocl_image1d_wo_t'
IMAGETY_WRITE_3_DIM_IMAGE
^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:178:3: note: expanded from macro 'IMAGETY_WRITE_3_DIM_IMAGE'
  IMAGETY_DEFINE(1, write, wo, image, ) \
  ^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:169:18: note: expanded from macro 'IMAGETY_DEFINE'
    using type = __ocl_image##dim##d_##ifarray_##amsuffix##_t;      \
                 ^
<scratch space>:18:1: note: expanded from here
__ocl_image1d_wo_t
^
In file included from /export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp:3:
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:191:1: error: unknown type name '__ocl_image2d_wo_t'
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:179:3: note: expanded from macro 'IMAGETY_WRITE_3_DIM_IMAGE'
  IMAGETY_DEFINE(2, write, wo, image, ) \
  ^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:169:18: note: expanded from macro 'IMAGETY_DEFINE'
    using type = __ocl_image##dim##d_##ifarray_##amsuffix##_t;      \
                 ^
<scratch space>:22:1: note: expanded from here
__ocl_image2d_wo_t
^
In file included from /export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp:3:
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:191:1: error: unknown type name '__ocl_image3d_wo_t'
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:180:3: note: expanded from macro 'IMAGETY_WRITE_3_DIM_IMAGE'
  IMAGETY_DEFINE(3, write, wo, image, )
  ^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:169:18: note: expanded from macro 'IMAGETY_DEFINE'
    using type = __ocl_image##dim##d_##ifarray_##amsuffix##_t;      \
                 ^
<scratch space>:26:1: note: expanded from here
__ocl_image3d_wo_t
^
In file included from /export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp:3:
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:193:1: error: unknown type name '__ocl_image1d_array_ro_t'
IMAGETY_READ_2_DIM_IARRAY
^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:183:3: note: expanded from macro 'IMAGETY_READ_2_DIM_IARRAY'
  IMAGETY_DEFINE(1, read, ro, image_array, array_) \
  ^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:169:18: note: expanded from macro 'IMAGETY_DEFINE'
    using type = __ocl_image##dim##d_##ifarray_##amsuffix##_t;      \
                 ^
<scratch space>:31:1: note: expanded from here
__ocl_image1d_array_ro_t
^
In file included from /export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp:3:
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:193:1: error: unknown type name '__ocl_image2d_array_ro_t'
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:184:3: note: expanded from macro 'IMAGETY_READ_2_DIM_IARRAY'
  IMAGETY_DEFINE(2, read, ro, image_array, array_)
  ^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:169:18: note: expanded from macro 'IMAGETY_DEFINE'
    using type = __ocl_image##dim##d_##ifarray_##amsuffix##_t;      \
                 ^
<scratch space>:36:1: note: expanded from here
__ocl_image2d_array_ro_t
^
In file included from /export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp:3:
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:194:1: error: unknown type name '__ocl_image1d_array_wo_t'
IMAGETY_WRITE_2_DIM_IARRAY
^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:187:3: note: expanded from macro 'IMAGETY_WRITE_2_DIM_IARRAY'
  IMAGETY_DEFINE(1, write, wo, image_array, array_) \
  ^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:169:18: note: expanded from macro 'IMAGETY_DEFINE'
    using type = __ocl_image##dim##d_##ifarray_##amsuffix##_t;      \
                 ^
<scratch space>:41:1: note: expanded from here
__ocl_image1d_array_wo_t
^
In file included from /export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/reqd-sub-group-size-host.cpp:3:
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:194:1: error: unknown type name '__ocl_image2d_array_wo_t'
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:188:3: note: expanded from macro 'IMAGETY_WRITE_2_DIM_IARRAY'
  IMAGETY_DEFINE(2, write, wo, image_array, array_)
  ^
/export/iusers/smanna/my_sycl_ws/llvm/clang/test/CodeGenSYCL/Inputs/sycl.hpp:169:18: note: expanded from macro 'IMAGETY_DEFINE'
    using type = __ocl_image##dim##d_##ifarray_##amsuffix##_t;      \
                 ^
<scratch space>:46:1: note: expanded from here
__ocl_image2d_array_wo_t
^
10 errors generated.

Reply:
test is not updated since mock header does not work on host.